src/modules/navigator/mission_feasibility_checker.cpp

```diff
@@ -200,33 +214,33 @@ bool MissionFeasibilityChecker::checkHomePositionAltitude(dm_item_t dm_current,
 		}
 
 		/* reject relative alt without home set */
-		if (missionitem.altitude_is_relative && !home_valid && isPositionCommand(missionitem.nav_cmd)) {
+		if (missionitem.altitude_is_relative && !home_valid && MissionBlock::item_contains_position(&missionitem)) {
 
 			warning_issued = true;
 
 			if (throw_error) {
-				mavlink_log_critical(_mavlink_log_pub, "Rejecting mission: No home pos, WP %d uses rel alt", i + 1);
+				mavlink_log_critical(_navigator->get_mavlink_log_pub(), "Mission rejected: No home pos, WP %d uses rel alt", i + 1);
 				return false;
 
 			} else	{
-				mavlink_log_critical(_mavlink_log_pub, "Warning: No home pos, WP %d uses rel alt", i + 1);
+				mavlink_log_critical(_navigator->get_mavlink_log_pub(), "Warning: No home pos, WP %d uses rel alt", i + 1);
 				return true;
 			}
 		}
 
 		/* calculate the global waypoint altitude */
 		float wp_alt = (missionitem.altitude_is_relative) ? missionitem.altitude + home_alt : missionitem.altitude;
 
-		if (home_alt > wp_alt && isPositionCommand(missionitem.nav_cmd)) {
+		if ((home_alt > wp_alt) && MissionBlock::item_contains_position(&missionitem)) {
```

Comment:
This can change the behavior because it's now a blacklist instead of a whitelist. I'm not sure of all the implications but if you have tested it, that probably needs to be enough.

https://github.com/PX4/Firmware/blob/809fec8c0579835d1fa9f1315f7ebc4398800502/src/modules/navigator/mission_block.cpp#L491